src/test/java/pl/mantiscrab/budgetr/registration/RegistrationTest.java

```diff
@@ -0,0 +1,72 @@
+package pl.mantiscrab.budgetr.registration;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+import org.springframework.web.util.UriComponentsBuilder;
+import pl.mantiscrab.budgetr.registration.dto.UserDto;
+import pl.mantiscrab.budgetr.registration.dto.UserRegisterDto;
+
+import java.net.URI;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder.on;
+import static pl.mantiscrab.budgetr.registration.TestUserData.sampleRegisterDto;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+class RegistrationTest {
+    private static final String LOCALHOST = "http://localhost";
+    @LocalServerPort
+    private int port;
+    @Autowired
+    TestRestTemplate restTemplate;
+
+    @Test
+    void registrationTest() {
```

Comment:
The test method could be description

E.g.
shouldRegisterUserWhenNotExists

Or with description